test/integration/policyConsumptionTest.ts

```diff
@@ -51,42 +52,37 @@ describe("Test: that policy lists are consumed by the associated synapse module"
         if(this.config.web.ruleServer.enabled) {
             this.timeout(5000)
             LogService.debug('policyConsumptionTest', `Rules at end of test ${JSON.stringify(await currentRules(this.mjolnir), null, 2)}`);
-            const mjolnir = config.RUNTIME.client!;
             // Clear any state associated with the account.
-            await mjolnir.setAccountData('org.matrix.mjolnir.watched_lists', {
+            await this.mjolnir.client.setAccountData('org.matrix.mjolnir.watched_lists', {
                 references: [],
             });
         }
     })
     this.beforeAll(async function() {
-        if (!this.config.web.ruleServer.enabled) {
+        let config = configRead();
+        if (!config?.web?.ruleServer?.enabled) {
             LogService.warn("policyConsumptionTest", "Skipping policy consumption test because the ruleServer is not enabled")
             this.skip();
         }
     })
-    this.beforeEach(async function () {
-        this.timeout(1000);
-        const mjolnir = this.config.RUNTIME.client!;
-    })
     it('blocks users in antispam when they are banned from sending messages and invites serverwide.', async function() {
         this.timeout(20000);
         // Create a few users and a room.
         let badUser = await newTestUser(this.config.homeserverUrl, { name: { contains: "spammer" }});
         let badUserId = await badUser.getUserId();
-        const mjolnir = config.RUNTIME.client!
-        let mjolnirUserId = await mjolnir.getUserId();
+        let mjolnirUserId = await this.mjolnir.client.getUserId();
         let moderator = await newTestUser(this.config.homeserverUrl, { name: { contains: "moderator" }});
         this.moderator = moderator;
         await moderator.joinRoom(this.mjolnir.managementRoomId);
         let unprotectedRoom = await badUser.createRoom({ invite: [await moderator.getUserId()]});
         // We do this so the moderator can send invites, no other reason.
         await badUser.setUserPowerLevel(await moderator.getUserId(), unprotectedRoom, 100);
         await moderator.joinRoom(unprotectedRoom);
-        const banList = await createBanList(this.mjolnir.managementRoomId, mjolnir, moderator);
+        const banList = await createBanList(this.mjolnir.managementRoomId, this.mjolnir.client, moderator);
         await badUser.sendMessage(unprotectedRoom, {msgtype: 'm.text', body: 'Something bad and mean'});
 
-        await waitForRuleChange(this.config.web.address, this.mjolnir.config.web.port, async () => {
```

Comment:
I think you have forgot to add the change to `waitForRuleChange`?

Reply:
Turns out that `waitForRuleChange` had already been refactored, just not its call sites.